src/Storages/ExternalStream/Kafka/KafkaSink.cpp

```diff
@@ -167,13 +167,17 @@ KafkaSink::KafkaSink(const Kafka * kafka, const Block & header, ContextPtr conte
     /// properies from settings have higher priority
     producer_params.emplace_back("bootstrap.servers", kafka->brokers());
     producer_params.emplace_back("security.protocol", kafka->securityProtocol());
-    if (boost::iequals(kafka->securityProtocol(), "SASL_SSL"))
+    if (boost::iequals(kafka->securityProtocol(), "SASL_PLAINTEXT")
+        || boost::iequals(kafka->securityProtocol(), "SASL_SSL"))
     {
         producer_params.emplace_back("sasl.mechanisms", "PLAIN");
         producer_params.emplace_back("sasl.username", kafka->username());
         producer_params.emplace_back("sasl.password", kafka->password());
     }
 
+    if (boost::iequals(kafka->securityProtocol(), "SASL_SSL"))
+        producer_params.emplace_back("ssl.ca.location", kafka->sslCaCertFile());
```

Comment:
The CA file should be optional, for example, if the target Kafka cluster uses a well-known CA certificated cert, there is no need to specify it.